### PR TITLE
Add 334 release notes

### PR DESCRIPTION
**Pull request by martint**
Fixes: https://github.com/prestosql/presto/issues/3626

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+* Fix malformed reads when caching parallel warmup is enabled. (:issue:`3772`)
+* Fix evicting of cached data while still under size eviction threshold. (:issue:`3772`)
+* Improve performance when creating unpartitioned external tables over large data sets. (:issue:`3624`)
+* Leverage Parquet file statistics when reading decimal columns. (:issue:`3581`)
+* Change type of ``$file_modified_time`` hidden column from ``bigint`` to ``timestamp with timezone type``. (:issue:`3611`)
+* Add caching support for HDFS and Azure. (:issue:`3772`)
+* Fix S3 connection pool depletion when caching parallel warmup is enabled. (:issue:`3772`)
+* Don't initialize cache on coordinator by default. (:issue:`3820`)
+* Use caching asynchronous read mode by default. (:issue:`3799`)
+* Add session property for disabling Hive cache. (:issue:`3843`)
```
[the remark this answers is not in the document]
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-334.rst:66]: It's `cache_enabled`

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+Release 333 (XX May 2020)
+=========================
+
+General Changes
+---------------
+
+* Fix incorrect query results for certain queries involving comparisons of ``real`` and ``double`` types
+  when values include negative zero. (:issue:`3745`)
+* Fix late materialization with aggregation queries. (:issue:`3577`)
+* Fix late materialization with empty splits. (:issue:`3577`)
```
[the remark this answers is not in the document]
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-334.rst:11]: When table is empty query would fail

[the remark this answers is not in the document]
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-334.rst]: Is that the same for both items?

[the remark this answers is not in the document]
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-334.rst]: > Is that the same for both items?

yes.

File: presto-main/src/test/java/io/prestosql/sql/query/TestLateral.java [hunk 3]
```diff
@@ -67,6 +67,6 @@ public void testUncorrelatedLateral()
     @Test
     public void testNotInScope()
     {
-        assertions.assertFails("SELECT * FROM (VALUES 1) t(a), (SELECT * FROM LATERAL (SELECT a))", "line 1:63: Column 'a' cannot be resolved");
+        assertions.assertFails("SELECT * FROM (V3829ALUES 1) t(a), (SELECT * FROM LATERAL (SELECT a))", "line 1:63: Column 'a' cannot be resolved");
```
Comment (losipiuk) [inline presto-main/src/test/java/io/prestosql/sql/query/TestLateral.java:70]: seems incidental :)

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-334.rst:68]: this line needs to be removed after: https://github.com/prestosql/presto/pull/3875

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 4]
```diff
@@ -0,0 +1,87 @@
+=========================
+Release 333 (XX May 2020)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:2]: 334

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+* Fix failure when querying an empty table with late materialization enabled. (:issue:`3577`)
+* Fix failure when the inputs to ``UNNEST`` are repeated. (:issue:`3587`)
+* Fix failure when an aggregation is used in the arguments to :func:`format`. (:issue:`3829`)
+* Fix :func:`localtime` and :func:`current_time` for session zones with DST or with historical offset changes
+  in legacy (default) timestamp semantics. (:issue:`3846`, :issue:`3850`)
+* Improve performance of queries involving :func:`row_number`. (:issue:`3614`)
+* Improve performance of queries containing ``LIKE`` predicate. (:issue:`3618`)
+* Improve query performance when dynamic filtering is enabled. (:issue:`3632`)
+* Improve performance for queries that read fields from nested structures. (:issue:`2672`)
+* Add variant of :func:`random` function that produces a number between m and n. (:issue:`1848`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:19]:
```suggestion
* Add variant of :func:`random` function that produces a number between given two numbers. (:issue:`1848`)
```

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+* Fix dynamic filter failures in complex spatial join queries. (:issue:`3694`)
+* Add :doc:`/connector/oracle`. (:issue:`1959`)
+* Add :doc:`/connector/pinot`. (:issue:`2028`)
+* Add support for modern HTTP forwarded headers. Processing of HTTP forwarded headers are now controlled by the
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:23]: "modern" is not descriptive

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+CLI Changes
+-----------
+
+* Improve rendering of elapsed time for short queries. (:issue:`3311`)
+
+Web UI Changes
+--------------
+
+* Add fixed, certificate, JWT and Kerberos to UI authentication. (:issue:`3433`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:35]: maybe `fixed` -> ``` ``fixed`` ```, since it's a used a name, not an adjective?

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+* Fix failure when executing ``SHOW CREATE TABLE``. (:issue:`3718`)
+* Improve performance for ``count(*)`` queries. (:issue:`3512`)
+* Add support for raw Elasticsearch queries. (:issue:`3735`)
+
+Hive Connector Changes
+----------------------
+
+* Fix creation of external tables using ``CREATE TABLE AS``. Previously such table was created
+  as managed and with default location. (:issue:`3755`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:55]: This was introduced recently (330). Do we wa want to mention when?

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+* Fix incorrect table statistics for new external table. (:issue:`3819`)
+* Prevent Presto from starting when cache fails to initialize. (:issue:`3749`)
+* Fix race condition that could cause caching to be unwillingly disabled. (:issue:`3729`, :issue:`3810`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:58]:
```suggestion
* Fix race condition that could cause caching to be permanently disabled. (:issue:`3729`, :issue:`3810`)
```

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+* Fix malformed reads when caching parallel warmup is enabled. (:issue:`3772`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:59]: parallel warmup -> async mode

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+* Fix evicting of cached data while still under size eviction threshold. (:issue:`3772`)
+* Improve performance when creating unpartitioned external tables over large data sets. (:issue:`3624`)
+* Leverage Parquet file statistics when reading decimal columns. (:issue:`3581`)
+* Change type of ``$file_modified_time`` hidden column from ``bigint`` to ``timestamp with timezone type``. (:issue:`3611`)
+* Add caching support for HDFS and Azure. (:issue:`3772`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:64]: Azure -> Azure storages 

(or something like that)

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+* Fix S3 connection pool depletion when caching parallel warmup is enabled. (:issue:`3772`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:65]: parallel warmup  -> async mode

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+* Don't initialize cache on coordinator by default. (:issue:`3820`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:66]: by default, or never?
i understand there is no caching on coordinator whatsoever?

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+* Use caching asynchronous read mode by default. (:issue:`3799`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:67]: asynchronous read mode -> async mode ?? how do we want to call it?

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-334.rst:68]: (if this stays) disabling -> skipping?

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+* Add session property for disabling Hive cache. (:issue:`3843`)
+* Allow caching delegation token for Hive thrift metastore. This can be configured with
+  the ``hive.metastore.thrift.delegation-token.cache-ttl`` and ``hive.metastore.thrift.delegation-token.cache-maximum-size``
+  configuration properties. (:issue:`3771`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:71]: This is enabled by default, so "Allow caching" -> "Cache"

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+
+MemSQL Connector Changes
+------------------------
+
+* Include :doc:`/connector/memsql` in the server build. (:issue:`3743`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:76]:
```suggestion
* Include :doc:`/connector/memsql` in the server build.
  Previously the connector was not part of the project's release artifacts. (:issue:`3743`)
```

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+
+MongoDB Connector Changes
+-------------------------
+
+* Allow case insensitive database and collection name. This can be enabled with the
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-334.rst:81]:
```suggestion
* Support case insensitive database and collection names. This can be enabled with the
```

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 5 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-334.rst:19]: How about

> that produces a number in the provided range

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 6 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-334.rst:24]:
```
standard compliant (:rfc:`7239`)

Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-334.rst:24]: are -> is

"Processing of ... is controlled by"

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+* Improve performance of queries containing ``LIKE`` predicate. (:issue:`3618`)
+* Improve query performance when dynamic filtering is enabled. (:issue:`3632`)
+* Improve performance for queries that read fields from nested structures. (:issue:`2672`)
+* Add variant of :func:`random` function that produces a number between m and n. (:issue:`1848`)
+* Fix dynamic filter failures in complex spatial join queries. (:issue:`3694`)
+* Add :doc:`/connector/oracle`. (:issue:`1959`)
+* Add :doc:`/connector/pinot`. (:issue:`2028`)
+* Add :doc:`/connector/prometheus`. (:issue:`2321`)
+* Add support for modern HTTP forwarded headers. Processing of HTTP forwarded headers are now controlled by the
+  ``http-server.process-forwarded`` configuration, and the old ``http-server.authentication.allow-forwarded-https`` and
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-334.rst:25]: configuration property

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 19]
```diff
@@ -0,0 +1,95 @@
...
+  ``dispatcher.forwarded-header`` configurations are no longer supported. (:issue:`3714`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-334.rst:26]: configuration properties

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 20]
```diff
@@ -0,0 +1,95 @@
...
+* Add pluggable :doc:`/develop/certificate-authenticator`. (:issue:`3804`)
+* Show distributed plan by default in :doc:`/sql/explain`. (:issue:`3724`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-334.rst:28]: Move this above, before the Add items

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+* Fix failure when the inputs to ``UNNEST`` are repeated. (:issue:`3587`)
+* Fix failure when an aggregation is used in the arguments to :func:`format`. (:issue:`3829`)
+* Fix :func:`localtime` and :func:`current_time` for session zones with DST or with historical offset changes
+  in legacy (default) timestamp semantics. (:issue:`3846`, :issue:`3850`)
+* Improve performance of queries involving :func:`row_number`. (:issue:`3614`)
+* Improve performance of queries containing ``LIKE`` predicate. (:issue:`3618`)
+* Improve query performance when dynamic filtering is enabled. (:issue:`3632`)
+* Improve performance for queries that read fields from nested structures. (:issue:`2672`)
+* Add variant of :func:`random` function that produces a number between m and n. (:issue:`1848`)
+* Fix dynamic filter failures in complex spatial join queries. (:issue:`3694`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-334.rst:20]: Move above with the other Fix

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 7 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-334.rst:43]: Add comma after JWT

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+Web UI Changes
+--------------
+
+* Add fixed, certificate, JWT and Kerberos to UI authentication. (:issue:`3433`)
+* Show join distribution type in Live Plan. (:issue:`1323`)
+
+JDBC Driver Changes
+-------------------
+
+* Improve performance of ``DatabaseMetaData#getColumns`` method in case when parameters
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-334.rst:49]:
```
* Improve performance of ``DatabaseMetaData.getColumns()`` when the
  parameters contain unescaped ``%`` or ``_``. (:issue:`1620`)

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 23]
```diff
@@ -0,0 +1,95 @@
...
+  contain unescaped ``%`` or ``_``. (:issue:`1620`)
+
+Elasticsearch Connector Changes
+-------------------------------
+
+* Fix failure when executing ``SHOW CREATE TABLE``. (:issue:`3718`)
+* Improve performance for ``count(*)`` queries. (:issue:`3512`)
+* Add support for raw Elasticsearch queries. (:issue:`3735`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-334.rst:57]: Do we explain this somewhere?

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+
+Hive Connector Changes
+----------------------
+
+* Fix matching bucket filenames without leading zeros. (:issue:`3702`)
+* Fix creation of external tables using ``CREATE TABLE AS``. Previously such table was created
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-334.rst:63]:
```
* Fix creation of external tables using ``CREATE TABLE AS``. Previously, the
  tables were created as managed and with the default location. (:issue:`3755`)

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 25]
```diff
@@ -0,0 +1,95 @@
...
+  as managed and with default location. (:issue:`3755`)
+* Fix incorrect table statistics for new external table. (:issue:`3819`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-334.rst:65]: > Fix incorrect table statistics for newly created external tables.

File: presto-docs/src/main/sphinx/release/release-334.rst [hunk 26]
```diff
@@ -0,0 +1,95 @@
...
+* Prevent Presto from starting when cache fails to initialize. (:issue:`3749`)
+* Fix race condition that could cause caching to be unwillingly disabled. (:issue:`3729`, :issue:`3810`)
+* Fix malformed reads when caching parallel warmup is enabled. (:issue:`3772`)
+* Fix evicting of cached data while still under size eviction threshold. (:issue:`3772`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-334.rst:69]: > Fix eviction of cached data while still under size eviction threshold.

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 11 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-334.rst:73]: > Azure file systems

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 16 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-334.rst:84]: "release artifacts" is not correct. I think we can just remove the second sentence.

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 7 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-334.rst:43]: Just quote "fixed" but not the others? That looks inconsistent.

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 8 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-334.rst:64]: What was introduced recently? The bug? Or the ability to create external tables?

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 13 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-334.rst:75]: @sopel39 ?

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 16 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-334.rst:84]: How about:

> Include ... in the server packaging.

or

> Include ... in the server tarball and RPM

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 23 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-334.rst:57]: No, we don't have documentation for this yet.

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 13 above)
Comment (sopel39) [reply presto-docs/src/main/sphinx/release/release-334.rst:75]: Let me rephrase:
```
Disable caching on coordinator by default. It can still be enabled by feature toggle for development purposes. (:issue:`3820`)
```

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 8 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-334.rst]: The bug _and_ ability to create external tables with data.

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 13 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-334.rst:75]: Let's drop the second sentence. "disable by default" implies it can be enabled with config, and we don't need to mention why

> Disable caching on coordinator by default.

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 14 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-334.rst:76]: The config is `hive.cache.read-mode=ASYNC` so this sounds fine. Maybe flip the sentence

> Use asynchronous read mode for caching by default.

File: presto-docs/src/main/sphinx/release/release-334.rst (hunk 16 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-334.rst:84]: The second one looks good
